### DISCUSSION TITLE
Hello, how the useQuery work with react rendering?

### DISCUSSION THREAD

**Discussion by vide89** [urql-graphql/urql#2304] [Q&A] [answered]
hello, really thanks for the lib, especially the docs it's very clear for someone who just start w/ graphql
I'm new with graphql, and want to fetch one row data periodically
urql provide query with poll interval and subscription that suitable with what I need
tried with the useQuery like these
```
// initialFetch this will trigger when the component should start send the poll request, 
// initialFetch set happened in the deeper level of child components
const [result, reexecuteQuery] = useQuery({
    query: getLatestUserCreated,
    context: {
      ...urqlContextOptions,
      pause: !initialFetch
      requestPolicy: 'cache-and-network',
      pollInterval: 200000
    }
  }) 
  ```
the component is on the upper layer of whole app layout, when I attach to the component my app looping and crashed, checking with console.log it loop way too much and poll interval seems doesn't have any effect (can see from the network tab it send many)
is there any way to solve this?

**Comment by JoviDeCroock** [comment, marked as answer]
We tell people to use `useMemo` for context as we can't predict when a consumer wants to refetch their query on changing context values. This can be seen [here](https://formidable.com/open-source/urql/docs/basics/react-preact/#context-options)

Furthermore some remarks:
- `pollInterval` has been deprecated, docs on how to replace are [here](https://formidable.com/open-source/urql/docs/basics/react-preact/#reexecuting-queries)
- `pause` belongs on the main object next to [`query`](https://formidable.com/open-source/urql/docs/basics/react-preact/#pausing-usequery)

An example:

```js
const [result, reexecuteQuery] = useQuery({
    query: getLatestUserCreated,
     pause: !initialFetch
    context: useMemo(() => ({
      ...urqlContextOptions,
      requestPolicy: 'cache-and-network',
    }), [])
  }) 
```

**Reply by vide89** [reply]
I see, can I ask one more thing about subscription? sorry I'm still new
So the case like these, user can joined multiple rooms and other users in same room can see notification how many total users in the room
If user stay in a room, the subscription working 
only after user switch room it's still fetch the same room info, confirmed from console the room id still same
I'm not sure do I need to unsubscribe & subscribe again?

**Reply by JoviDeCroock** [reply]
It depends, when your variables change for the subscription then it will do all of that for you, if they don't it will keep the current one open

**Reply by vide89** [reply]
the subscription not using any variable, then how can I close it? 
on the docs only find pause option https://github.com/FormidableLabs/urql/blob/urql%401.9.8/docs/api/urql.md

**Reply by JoviDeCroock** [reply]
Yes turning pause to true or unmounting will stop it

**Reply by vide89** [reply]
Ohh now it works, thanks a lot Jovi :+1:

**Reply by vide89** [reply]
hi Jovi, sorry to open this discussion again
today I tried with several browsers to simulate multiple users enters one room
I noticed, the subscription query have correct variable changed from urql devtools 
but the query result is incorrect
Sample:
I only switch room with one user, 
   - enter room A (initial users 0, current users 1, correct)
   - switch to room B (initial users 0, current users 1, correct)

but when I tried switching 3 rooms, 
   - enter room A (initial users 1, current users 2, correct)
   - switch to room B (initial users 0, current users 1, correct) 
   - switch to room C (initial users 2, current users 1, incorrect)
but if I'm refresh the browser and start from room C now it show correctly (3 users)
do you might know what's the possible cause?
